src/org/openlcb/cdi/swing/CdiPanel.java

```diff
@@ -1010,10 +1058,206 @@ public void propertyChange(PropertyChangeEvent event) {
+                   writeEntry(e.key, e.getValue());
+                }
+
+                @Override
+                public void visitInt(ConfigRepresentation.IntegerEntry e) {
+                   writeEntry(e.key, Long.toString(e.getValue()));
+                }
+
+                @Override
+                public void visitEvent(ConfigRepresentation.EventEntry e) {
+                   writeEntry(e.key, org.openlcb.Utilities.toHexDotsString(e.getValue
+                           ().getContents()));
+                }
+                
+                protected void writeEntry(String key, String entry) {
+                    result.append(key);
+                    result.append("=");
+                    // result.append(entry); // use the value currently in CD
+                    result.append(entriesByKey.get(key).getCurrentValue()); // use value currently in UI
+                    result.append("\n");
+                }
+            };
+            
+            visitor.visitGroupRep(rep);
+            return new String(result);
+        }  
+            
+        
+        /**
+         * Perform a "paste" operation into a selected group tab
+         */
+        protected void performGroupReplPaste(int index, ConfigRepresentation.GroupRep rep) {
+            // retrieve from clipboard
+            Clipboard c = Toolkit.getDefaultToolkit().getSystemClipboard();
+            Transferable t = c.getContents( null );
+            String newContentString = "";
+            if ( t.isDataFlavorSupported(DataFlavor.stringFlavor) ) {
+                try {
+                    newContentString = (String)t.getTransferData( DataFlavor.stringFlavor );
+                } catch (UnsupportedFlavorException | IOException e) {
+                    // this can never happen as we checked bafore
+                    return;
+                }
+            } // this should always have succeeded, but if it doesn't the match below will fail
+            
+            // store the values that are going to be replaced
+            String previousContentString = groupReplToString(rep); 
+
+            String[] newContentLines = newContentString.split("\n");
+            String[] previousContentLines = previousContentString.split("\n");
+            
+            // compare keys to see if the variables match up
+            // this version just checks the line count, more could be added here
+            if (previousContentLines.length != newContentLines.length) {
+                logger.log(Level.WARNING, "Cannot paste into a mis-matching entry type");
+                return;
+            }
+            
+            // change the repl number in the newContentLines to this index
+            // First, find the prefix we're going to change
+            List<String> list = java.util.Arrays.asList(newContentLines);
+            String prefix = Utilities.longestLeadingSubstring(list);
+            // That prefix should end with "(1)."
```

Comment:
Not sure of that.  Is this non-replicated group in the RR-CirKits Tower LCC the kind of thing you were thinking about?  It works OK.
```
  <segment space="251">
    <name>NODE ID</name>
    <group>
      <name>Your name and description for this node</name>
      <string size="63">
        <name>Node Name</name>
      </string>
      <string size="64" offset="1">
        <name>Node Description</name>
      </string>
    </group>
  </segment>
```

Reply:
Never mind, that's CDI clause above isn't in a tab.  Working on this more...

Reply:
This is resolved by the use of `rep.key` below.

Reply:
The counterexample I meant is when there is only one field within the repeated group:
```
<group replication="4"><name>something</name> <repname>foo</repname>
<int size=2><name>hello</name></int>
</group>
```

as here the longest prefix will be the entire first key, i.e. `something.entry(1).hello`

It gets even nastier when we have a single nested group below a replicated group.
```
<group replication="4"><name>something</name> <repname>foo</repname>
<group><name>else</name>
<int size=2><name>hello</name></int>
<string size=20><name>world</name></int>
</group>
</group>
``` 

because here the copy-paste items are going to be like
```
something.entry(1).else.hello
something.entry(1).else.world
```

I don't have a very good suggestion on how to fix this. Ideal would be if we could know what the rep.key is on the other side, when it was copied. Barring that we could do something with counting the number of dotted components. We are looking for the following property:
- what we cut off must be a common prefix (=> prefix of the longest common prefix)
- The thing we cut off ends with ")."
- when we take the remaining lines after the cut-off, the average number of dots per line is the same as the average number of dots in the previousContentLines, when we take off rep.key as prefix from each line.

It's probably OK to not block this PR on resolving this problem, but fix it in a followup. It already does work for the most common cases (copying inputs and outputs).

Reply:
Maybe the simplest fix is to cut back `prefix` to end with `")."`